libs/testing/tests/unit/test_macros.cpp

```diff
@@ -0,0 +1,82 @@
+//  Copyright (c) 2020 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/hpx_main.hpp>
+#include <hpx/testing.hpp>
+
+#include <sstream>
+
+int main(int argc, char* argv[])
+{
+    std::stringstream strm;
+
+    // testing macros
+    HPX_TEST(0 == 0);
+    HPX_TEST(strm, 0 == 0);
+
+    HPX_TEST_MSG(0 == 0, "should be true");
+    HPX_TEST_MSG(strm, 0 == 0, "should be true");
+
+    HPX_TEST_EQ(0, 0);
+    HPX_TEST_EQ(strm, 0, 0);
+
+    HPX_TEST_EQ_MSG(0, 0, "should be equal");
+    HPX_TEST_EQ_MSG(strm, 0, 0, "should be equal");
+
+    HPX_TEST_NEQ(0, 1);
+    HPX_TEST_NEQ(strm, 0, 1);
+
+    HPX_TEST_NEQ_MSG(0, 1, "should not be equal");
+    HPX_TEST_NEQ_MSG(strm, 0, 1, "should not be equal");
+
+    HPX_TEST_LT(0, 1);
+    HPX_TEST_LT(strm, 0, 1);
+
+    HPX_TEST_LT_MSG(0, 1, "should be less");
+    HPX_TEST_LT_MSG(strm, 0, 1, "should be less");
+
+    HPX_TEST_LTE(1, 1);
+    HPX_TEST_LTE(strm, 1, 1);
+
+    HPX_TEST_LTE_MSG(1, 1, "should be less equal");
+    HPX_TEST_LTE_MSG(strm, 1, 1, "should be less equal");
+
+    HPX_TEST_RANGE(1, 1, 1);
+    HPX_TEST_RANGE(strm, 1, 1, 1);
+
+    HPX_TEST_RANGE_MSG(1, 1, 1, "should be in range");
+    HPX_TEST_RANGE_MSG(strm, 1, 1, 1, "should be in range");
+
+    // sanity macro tests
+    HPX_SANITY(0 == 0);
+    HPX_SANITY(strm, 0 == 0);
+
+    HPX_SANITY_MSG(0 == 0, "should be true");
+    HPX_SANITY_MSG(strm, 0 == 0, "should be true");
+
+    HPX_SANITY_EQ(0, 0);
+    HPX_SANITY_EQ(strm, 0, 0);
+
+    HPX_SANITY_EQ_MSG(0, 0, "should be equal");
+    HPX_SANITY_EQ_MSG(strm, 0, 0, "should be equal");
+
+    HPX_SANITY_NEQ(0, 1);
+    HPX_SANITY_NEQ(strm, 0, 1);
+
+    HPX_SANITY_LT(0, 1);
+    HPX_SANITY_LT(strm, 0, 1);
+
+    HPX_SANITY_LTE(1, 1);
+    HPX_SANITY_LTE(strm, 1, 1);
+
+    HPX_SANITY_RANGE(1, 1, 1);
+    HPX_SANITY_RANGE(strm, 1, 1, 1);
+
+    // there shouldn't be any output being generated
+    HPX_TEST(strm.str().empty());
```

Comment:
Should there be the opposite test to check that the error messages actually get written to the stream? Perhaps not character-by-character exactly what we expect, but `strm` should at least not be empty if a test fails.